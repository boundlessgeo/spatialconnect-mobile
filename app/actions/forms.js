'use strict';
import * as sc from 'spatialconnect/native';

export const loadForms = () => {
  return dispatch => {
    sc.forms().subscribe(data => {
      dispatch({
        type: 'ADD_FORM_LIST',
        forms: data.forms
      });
    });
  };
};