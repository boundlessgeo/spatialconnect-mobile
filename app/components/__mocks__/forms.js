export default [{
  id: 1,
  version: 0,
  form_key: 'weed_inspector',
  form_label: 'Weed Inspector',
  fields: [{
    id: 1,
    type: 'number',
    field_label: 'Weed Stage',
    field_key: 'weed_stage',
    is_required: true,
    position: 0,
    is_integer: false
  }, {
    id: 2,
    type: 'counter',
    field_label: 'Number of Plants',
    field_key: 'number_of_plants',
    position: 1,
    initial_value: '0'
  }, {
    id: 3,
    type: 'slider',
    field_label: 'Infestation Size',
    field_key: 'infestation_size',
    position: 2,
    initial_value: '0',
    minimum: '0',
    maximum: '100'
  }, {
    id: 4,
    type: 'counter',
    field_label: 'Degree',
    field_key: 'degree',
    is_required: true,
    position: 3,
    initial_value: '0',
    minimum: '0'
  }, {
    id: 5,
    type: 'slider',
    field_label: 'Density',
    field_key: 'density',
    position: 4,
    initial_value: '0',
    minimum: '0',
    maximum: '100'
  }, {
    id: 6,
    type: 'string',
    field_label: 'Source',
    field_key: 'source',
    is_required: true,
    position: 5,
    pattern: ''
  }, {
    id: 7,
    type: 'date',
    field_label: 'Data Discovered',
    field_key: 'date_discovered',
    position: 6
  }, {
    id: 8,
    type: 'string',
    field_label: 'Common Name',
    field_key: 'common_name',
    is_required: true,
    position: 7,
    pattern: ''
  }, {
    id: 9,
    type: 'string',
    field_label: 'Genus',
    field_key: 'genus',
    is_required: true,
    position: 8,
    pattern: ''
  }, {
    id: 10,
    type: 'string',
    field_label: 'Species',
    field_key: 'species',
    position: 9,
    pattern: ''
  }, {
    id: 11,
    type: 'string',
    field_label: 'Scientific Name',
    field_key: 'scientific_name',
    position: 10,
    pattern: ''
  }, {
    id: 12,
    type: 'string',
    field_label: 'Class',
    field_key: 'class',
    position: 11,
    pattern: ''
  }]
}, {
  id: 2,
  version: 0,
  form_key: 'baseball_team',
  form_label: 'Baseball Team',
  fields: [{
    id: 13,
    type: 'string',
    field_label: 'Favorite?',
    field_key: 'team',
    position: 0
  }, {
    id: 14,
    type: 'string',
    field_label: 'Why?',
    field_key: 'why',
    position: 1
  }]
}, {
  id: 5,
  version: 3,
  form_key: 'bird_counter',
  form_label: 'Bird Counter',
  fields: [{
    id: 18,
    type: 'counter',
    field_label: 'Bird Count',
    field_key: 'bird_count',
    is_required: true,
    position: 1,
    initial_value: '0'
  }, {
    id: 17,
    type: 'string',
    field_label: 'Bird Species',
    field_key: 'bird_species',
    is_required: true,
    position: 0,
    pattern: ''
  }]
}];