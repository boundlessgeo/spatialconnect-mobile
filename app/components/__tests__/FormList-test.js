/*global jest,describe,it,expect*/
import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import FormList from '../FormList';
import forms from 'forms';

describe('<FormList />', () => {

  it('renders correctly', () => {
    const tree = renderer.create(
      <FormList forms={forms} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

});