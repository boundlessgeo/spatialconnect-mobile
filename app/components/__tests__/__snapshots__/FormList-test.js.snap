exports[`<FormList /> renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#e4e7ec",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "center",
      "padding": 0
    }
  }>
  <ListView
    dataSource={
      ListViewDataSource {
        "_cachedRowCount": 3,
        "_dataBlob": Object {
          "s1": Array [
            Object {
              "fields": Array [
                Object {
                  "field_key": "weed_stage",
                  "field_label": "Weed Stage",
                  "id": 1,
                  "is_integer": false,
                  "is_required": true,
                  "position": 0,
                  "type": "number"
                },
                Object {
                  "field_key": "number_of_plants",
                  "field_label": "Number of Plants",
                  "id": 2,
                  "initial_value": "0",
                  "position": 1,
                  "type": "counter"
                },
                Object {
                  "field_key": "infestation_size",
                  "field_label": "Infestation Size",
                  "id": 3,
                  "initial_value": "0",
                  "maximum": "100",
                  "minimum": "0",
                  "position": 2,
                  "type": "slider"
                },
                Object {
                  "field_key": "degree",
                  "field_label": "Degree",
                  "id": 4,
                  "initial_value": "0",
                  "is_required": true,
                  "minimum": "0",
                  "position": 3,
                  "type": "counter"
                },
                Object {
                  "field_key": "density",
                  "field_label": "Density",
                  "id": 5,
                  "initial_value": "0",
                  "maximum": "100",
                  "minimum": "0",
                  "position": 4,
                  "type": "slider"
                },
                Object {
                  "field_key": "source",
                  "field_label": "Source",
                  "id": 6,
                  "is_required": true,
                  "pattern": "",
                  "position": 5,
                  "type": "string"
                },
                Object {
                  "field_key": "date_discovered",
                  "field_label": "Data Discovered",
                  "id": 7,
                  "position": 6,
                  "type": "date"
                },
                Object {
                  "field_key": "common_name",
                  "field_label": "Common Name",
                  "id": 8,
                  "is_required": true,
                  "pattern": "",
                  "position": 7,
                  "type": "string"
                },
                Object {
                  "field_key": "genus",
                  "field_label": "Genus",
                  "id": 9,
                  "is_required": true,
                  "pattern": "",
                  "position": 8,
                  "type": "string"
                },
                Object {
                  "field_key": "species",
                  "field_label": "Species",
                  "id": 10,
                  "pattern": "",
                  "position": 9,
                  "type": "string"
                },
                Object {
                  "field_key": "scientific_name",
                  "field_label": "Scientific Name",
                  "id": 11,
                  "pattern": "",
                  "position": 10,
                  "type": "string"
                },
                Object {
                  "field_key": "class",
                  "field_label": "Class",
                  "id": 12,
                  "pattern": "",
                  "position": 11,
                  "type": "string"
                }
              ],
              "form_key": "weed_inspector",
              "form_label": "Weed Inspector",
              "id": 1,
              "version": 0
            },
            Object {
              "fields": Array [
                Object {
                  "field_key": "team",
                  "field_label": "Favorite?",
                  "id": 13,
                  "position": 0,
                  "type": "string"
                },
                Object {
                  "field_key": "why",
                  "field_label": "Why?",
                  "id": 14,
                  "position": 1,
                  "type": "string"
                }
              ],
              "form_key": "baseball_team",
              "form_label": "Baseball Team",
              "id": 2,
              "version": 0
            },
            Object {
              "fields": Array [
                Object {
                  "field_key": "bird_count",
                  "field_label": "Bird Count",
                  "id": 18,
                  "initial_value": "0",
                  "is_required": true,
                  "position": 1,
                  "type": "counter"
                },
                Object {
                  "field_key": "bird_species",
                  "field_label": "Bird Species",
                  "id": 17,
                  "is_required": true,
                  "pattern": "",
                  "position": 0,
                  "type": "string"
                }
              ],
              "form_key": "bird_counter",
              "form_label": "Bird Counter",
              "id": 5,
              "version": 3
            }
          ]
        },
        "_dirtyRows": Array [
          Array [
            true,
            true,
            true
          ]
        ],
        "_dirtySections": Array [
          true
        ],
        "_getRowData": [Function defaultGetRowData],
        "_getSectionHeaderData": [Function defaultGetSectionHeaderData],
        "_rowHasChanged": [Function rowHasChanged],
        "_sectionHeaderHasChanged": [Function anonymous],
        "rowIdentities": Array [
          Array [
            "0",
            "1",
            "2"
          ]
        ],
        "sectionIdentities": Array [
          "s1"
        ]
      }
    }
    enableEmptySections={true}
    renderRow={[Function bound renderRow]}
    renderSeparator={[Function bound renderSeparator]}
    style={
      Object {
        "flex": 1
      }
    } />
</View>
`;
